src/DGtal/shapes/TriangulatedSurface.h

```diff
@@ -546,6 +550,77 @@ namespace DGtal
     */
     VertexRange allBoundaryVertices() const;
 
+    /**
+       @param a any arc.
+       @return either the four vertices of the two triangles bordering \a a, or the three vertices if this is an edge on the boundary.
+       @note O(1) operation
+       @note Order is, if arc a is (s,t), and a belongs to triangle (s,t,u) and opposite arc belongs to triangle (t,s,v): (t,u,s,v), (t,u,s), (t,s,v) 
+    */
+    VertexRange verticesAroundArc( const Arc a ) const;
+      
+    // ----------------------- Other services ---------------------------------
+  public:
+
+    /// An edge is (topologically) flippable iff: (1) it does not lie
+    /// on the boundary, (2) it is bordered by two triangles. (2) is
+    /// always true for a triangulated surface.
+    ///
+    /// @param a any arc.
+    /// @return 'true' if the edge containing \a a is topologically flippable.
+    bool isFlippable( const Arc a ) const;
+
+    /// Flip the edge associated to arc \a a.
+    ///
+    /// @param a any valid arc.
+    ///
+    /// @pre the edge must be flippable, `isFlippable( a ) == true`
+    /// @see isFlippable
+    ///
```

Comment:
maybe you could add the complexities as well for these elementary "Euler" operations..

Reply:
done